### PR TITLE
[core] Add cherry-pick `master` to `v6` action

### PR DESCRIPTION
**Pull request by LukasTy**
Add an action, which would cherry-pick a PR merged to master into a `v6.x` branch if it would have a `needs cherry-pick` label.

Could be useful for bug fixes during the v7 development / v6 maintenance mode.

### CODE REVIEW

File: .github/workflows/cherry-pick-master-to-v6.yml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+permissions: {}
+
+jobs:
+  cherry_pick_to_v6:
+    runs-on: ubuntu-latest
+    name: Cherry pick into v6
+    permissions:
+      pull-requests: write
+      contents: write
+    if: ${{ contains(github.event.pull_request.labels.*.name, 'needs cherry-pick') && contains(github.event.pull_request.labels.*.name, 'v6.x') && github.event.pull_request.merged == true }}
```
Comment (michelengelen) [inline .github/workflows/cherry-pick-master-to-v6.yml:18]: do we really need the `v6.x` label? Seems like 99.9% of PRs will be on `master` and when a cherry pick is needed we only add it to v6 anyways, right?

Comment (michelengelen) [reply .github/workflows/cherry-pick-master-to-v6.yml:18]: IMHO I would be in favor of that

Comment (LukasTy) [reply .github/workflows/cherry-pick-master-to-v6.yml]: Removed the extra check. 😉

**Review by michelengelen — APPROVED** [review]

**Review by cherniavskii — APPROVED** [review]

**Review by MBilalShafi — APPROVED** [review]
Thanks Lukas, I was thinking about backporting a couple of PRs, this workflow will help speed it up 🚀

**Pull request merged**